src/selfplay/loop.cc

```diff
@@ -132,6 +135,52 @@ void SelfPlayLoop::SendTournament(const TournamentInfo& info) {
   res += " draw " + std::to_string(info.results[1][0]) + " " +
          std::to_string(info.results[1][1]);
   SendResponse(res);
+
+  // Compute player1's elo relative to player 2.
+  auto wins = info.results[0][0]+info.results[0][1];
+  auto losses = info.results[2][0] + info.results[2][1];
+  auto draws = info.results[1][0] + info.results[1][1];
+  auto n = wins + losses + draws;
+  if (n == 0) {
+    // no games
+    return;
+  }
+  double w, l, d;
+  w = wins / (double)n;
+  l = losses / (double)n;
+  d = draws / (double)n;
+  // mu is the estimate of the expected score
+  auto mu = w + d / 2.0;
+  auto devW = w * pow(1.0 - mu, 2);
+  auto devL = l * pow(mu, 2);
+  auto devD = d * pow(0.5 - mu, 2);
+  // The variance of player1's score from a single game is
+  // approximated as devW + devL + devD; she scores
+  // 1 with probability ~w, 0.5 w.p. ~d, and 0 w.p. ~l.
+  // We approximate the variance of the average of n of these as
+  // (devW + devL + devD) / n by the central limit theorem
+  auto stddev = sqrt((devW + devL + devD) / n);
+  // The 95% confidence interval is thus given by
+  // (mu - stddev * 1.96, mu + stddev * 1.96)
+  auto muLow = mu - stddev * 1.96;
+  auto muHigh = mu + stddev * 1.96;
+  auto eloDelta = [](const double score) {
+    // Compute the difference in elo between players, for a given score
+    return -400. * log10(1.0 / score - 1.0);
+  };
+  auto inBounds = [](const double score) {
+    // is the score within the bounds of what we can estimate elo and elo bounds for?
+    return score > 0 && score < 1;
+  };
+  std::ostringstream out;
+  if (!inBounds(mu) || !inBounds(muLow) || !inBounds(muHigh)) {
+    return;
+  }
+  out << std::setprecision(3)
```

Comment:
Prepend that with `info string` at least. We are trying to have UCI-like protocol, so freeform text looks like an alien here.